tendrl/monitoring_integration/flows/delete_resource_from_graphite/__init__.py

```diff
@@ -0,0 +1,208 @@
+import etcd
+import os
+import datetime
+import ConfigParser
+
+
+from tendrl.commons import flows
+from tendrl.commons.utils import log_utils as logger
+from tendrl.commons.utils import etcd_utils
+from tendrl.monitoring_integration.grafana import create_dashboards
+
+
+
+class DeleteResourceFromGraphite(flows.BaseFlow):
+
+    def run(self):
+        super(UpdateGraphite, self).run()
+        cluster_id = self.parameters.get("TendrlContext.integration_id")
+        resource_name = str(
+            self.parameters.get("Trigger.resource_name")).lower()
+        resource_type = str(
+            self.parameters.get("Trigger.resource_type")).lower()
+        self.update_graphite(cluster_id, resource_name,
+                             resource_type.lower())
+
+    def get_data_dir_path(self):
+        carbon_path = "/etc/tendrl/monitoring-integration/carbon.conf"
+        if not os.path.exists(carbon_path):
+            return None
+        carbon_config = ConfigParser.ConfigParser()
+        carbon_config.read(carbon_path)
+        try:
+            whisper_path = str(carbon_config.get("cache", "local_data_dir")
+            whisper_path = os.path.join(whisper_path, "tendrl/")
+            return whisper_path
+        except KeyError:
+            return None
+
+
+    def update_graphite(self, cluster_id, resource_name, resource_type):
+        whisper_path = self.get_data_dir_path()
+        if not whisper_path:
+            logger.log("error", NS.get("publisher_id", None),
+                       {'message': "Cannot retrieve whisper path"})
+            return
+        if resource_type == "volume":
+            resource_path = os.path.join(WHISPER_PATH, "clusters",
+                                         str(cluster_id),
+                                         "volumes", resource_name)
+            archive_path = os.path.join(WHISPER_PATH, "clusters",
+                                        str(cluster_id),
+                                        "archive", "volumes")
+            if not os.path.exists(archive_path):
+                os.system("mkdir -p " + str(archive_path))
```

Comment:
os module has "makedirs" function, you can directly use that.
example: os.makedirs("a/b/c")